holoviews/plotting/comms.py

```diff
@@ -0,0 +1,165 @@
+import uuid
+
+import param
+from ipykernel.comm import Comm as IPyComm
+from IPython import get_ipython
+
+
+class Comm(param.Parameterized):
+    """
+    Comm encompasses any uni- or bi-directional connection between
+    a python process and a frontend allowing passing of messages
+    between the two. A Comms class must implement methods
+    send data and handle received message events.
+
+    If the Comm has to be set up on the frontend a template to
+    handle the creation of the comms channel along with a message
+    handler to process incoming messages must be supplied.
+
+    The template must accept three arguments:
+
+    * comms_target - A unique id to register to register as the comms target.
+    * msg_handler -  JS code that processes messages sent to the frontend.
+    * init_frame  -  The initial frame to render on the frontend.
+    """
+
+    template = ''
+
+    def __init__(self, plot, target=None, on_msg=None):
+        """
+        Initializes a Comms object
+        """
+        self.target = target if target else uuid.uuid4().hex
+        self._plot = plot
+        self._on_msg = on_msg
+        self._comm = None
+
+
+    def init(self, on_msg=None):
+        """
+        Initializes comms channel.
+        """
+
+
+    def send(self, data):
+        """
+        Sends data to the frontend
+        """
+
+
+    def decode(self, msg):
+        """
+        Decode incoming message, e.g. by parsing json.
+        """
+        return msg
+
+
+    @property
+    def comm(self):
+        if not self._comm:
+            raise ValueError('Comm has not been initialized')
+        return self._comm
+
+
+    def _handle_msg(self, msg):
+        """
+        Decode received message before passing it to on_msg callback
+        if it has been defined.
+        """
+        if self._on_msg:
+            self._on_msg(self.decode(msg))
+
+
+class SimpleJupyterComm(Comm):
+    """
```

Comment:
How about calling it ``JupyterPushComm``? From the sound of it, we may not need it in future one bokeh works with bi-directional comms.